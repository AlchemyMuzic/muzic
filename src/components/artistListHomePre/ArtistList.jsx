import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// import CharacterDetailsPresentation from './CharacterDetailsPresentation';

function ArtistList({ artists }) {

  const artistsElements = artists.map(artist => (
    <li key={artist.id}>
      <Link to={`/artists/${artist.id}/${artist.name}`}>
        <h2>{artist.name}</h2>
      </Link>
    </li>
  ));

  return (
    <ul data-testid="artistsTest">
      {artistsElements}
    </ul>
  );
}

ArtistList.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired
};

export default ArtistList;
