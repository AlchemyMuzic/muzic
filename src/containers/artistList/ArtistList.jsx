import React from 'react';
import ArtistList from '../../components/artistListHomePre/ArtistList';
import { useArtistList } from '../../services/hooks/useArtistList';
import SearchInput from '../../components/artistListHomePre/SearchInput';
import Loading from '../../components/loading/Loading';

const Home = () => {
  const { loading, artists, setSearch } = useArtistList();

  if(loading) return <Loading />;
  return <div>
    <SearchInput setSearch={setSearch}/> 
    <ArtistList artists={artists} />
  </div>;
};

export default Home;
