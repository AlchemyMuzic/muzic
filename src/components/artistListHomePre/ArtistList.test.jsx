import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import ArtistList from './ArtistList';

describe('ArtistList component', () => {
    afterEach(() => cleanup());
    it('renders a list of artists', () => {
        const { asFragment } = render(<BrowserRouter><ArtistList
            artists={[]} /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    })
})
