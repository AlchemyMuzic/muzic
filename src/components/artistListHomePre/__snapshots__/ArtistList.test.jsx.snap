// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders a list of artists 1`] = `
<DocumentFragment>
  <ul
    data-testid="artistsTest"
  />
</DocumentFragment>
`;
